k8s/k8s-workload-registrar/README.md

```diff
@@ -0,0 +1,716 @@
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: k8s-workload-registrar
+  namespace: spire
+data:
+  k8s-workload-registrar.conf: |
+    trust_domain = "example.org"
+    server_socket_path = "/tmp/spire-server/private/api.sock"
+    cluster = "example-cluster"
+    mode = "crd"
+    pod_annotation = "spiffe.io/spiffe-id"
+    metrics_bind_addr = "0"
+```
+
+Let's deploy the necessary files, including the base scenario plus the SPIFFE ID CRD definition, and examine the automatically created registration entries. Ensure that your working directory is `mode-crd`, and run:
+
+```console
+$ bash scripts/deploy-scenario.sh
+```
+
+Run the entry show command by executing:
+
+```console
+$ kubectl exec -it statefulset/spire-server -n spire -c spire-server -- bin/spire-server entry show -registrationUDSPath /tmp/spire-server/private/api.sock
+```
+
+The output should show the following registration entries:
+
+```console
+Found 3 entries
+Entry ID         : ...
+SPIFFE ID        : spiffe://example.org/k8s-workload-registrar/example-cluster/node/example-cluster-control-plane
+Parent ID        : spiffe://example.org/spire/server
+Revision         : 1
+TTL              : default
+Selector         : k8s_psat:agent_node_uid:08990bfd-3551-4761-8a1b-2e652984ffdd
+Selector         : k8s_psat:cluster:example-cluster
+
+Entry ID         : ...
+SPIFFE ID        : spiffe://example.org/testing/agent
+Parent ID        : spiffe://example.org/k8s-workload-registrar/example-cluster/node/example-cluster-control-plane
+Revision         : 1
+TTL              : default
+Selector         : k8s:node-name:example-cluster-control-plane
+Selector         : k8s:ns:spire
+Selector         : k8s:pod-uid:538886bb-48e1-4795-b386-10e97f50e34f
+DNS name         : spire-agent-jzc8w
+
+Entry ID         : ...
+SPIFFE ID        : spiffe://example.org/testing/example-workload
+Parent ID        : spiffe://example.org/k8s-workload-registrar/example-cluster/node/example-cluster-control-plane
+Revision         : 1
+TTL              : default
+Selector         : k8s:node-name:example-cluster-control-plane
+Selector         : k8s:ns:spire
+Selector         : k8s:pod-uid:78ed3fc5-4cff-476a-90f5-37d3abd47823
+DNS name         : example-workload-6877cd47d5-l4hv5
+```
+
+3 entries were created corresponding to the node, agent, and workload. For the node entry (the one that has the SPIRE Server SPIFFE ID as Parent ID), we see a difference in the selectors, comparing it with the selectors in the node entry created using Reconcile mode for the same pod: we find out that instead of using the node name, CRD mode stores the UID of the node where the agent is running on. As the node name is used in the SPIFFE ID assigned to the node, we can take this as a mapping from node UID to node name.
```

Comment:
```suggestion
Three entries were created corresponding to the node, agent, and workload. For the node entry (the one that has the SPIRE Server SPIFFE ID as Parent ID), we see a difference in the selectors compared to reconcile mode: instead of using the node name, CRD mode stores the UID of the node where the agent is running on. As the node name is used in the SPIFFE ID assigned to the node, we can use this as a mapping from node UID to node name.
```

Reply:
Here I tried to use "take" as a synonym of "interpret", because we don't actually _use_ this mapping.

Reply:
Ok, I see how my version is wrong but I don't quite understand the point you're making. We can use your updated version.

Reply:
I see this "mapping" thing occurring when the server issues the SVID to the agent. The server uses the selectors to know the identity that it should issue, so in this case, verifying the node UID that is present in the selectors, the server issues an SVID with an SPIFFE ID that has the node name in it. This is where I'm understanding that a mapping happens (node UID `08990bfd-....` -> node name `minikube`).